fendermint/vm/interpreter/src/fvm/broadcast.rs

```diff
@@ -49,87 +72,112 @@ where
             addr,
             gas_fee_cap,
             gas_premium,
+            max_retries: 0,
         }
     }
 
+    pub fn with_max_retries(mut self, max_retries: u8) -> Self {
+        self.max_retries = max_retries;
+        self
+    }
+
     pub async fn fevm_invoke(
         &self,
         contract: Address,
         calldata: et::Bytes,
         chain_id: ChainID,
     ) -> anyhow::Result<()> {
-        let sequence = self
-            .sequence()
-            .await
-            .context("failed to get broadcaster sequence")?;
+        let tx_hash = retry!(self.max_retries, {
+            let sequence = self
+                .sequence()
+                .await
+                .context("failed to get broadcaster sequence")?;
 
-        let factory = MessageFactory::new(self.secret_key.clone(), self.addr, sequence, chain_id);
+            let factory =
+                MessageFactory::new(self.secret_key.clone(), self.addr, sequence, chain_id);
 
-        // Using the bound client as a one-shot transaction sender.
-        let mut client = self.client.clone().bind(factory);
+            // Using the bound client as a one-shot transaction sender.
+            let mut client = self.client.clone().bind(factory);
 
-        // TODO: Maybe we should implement something like the Ethereum facade for estimating fees?
-        // I don't want to call the Ethereum API directly (it would be one more dependency).
-        // Another option is for Fendermint to recognise transactions coming from validators
-        // and always put them into the block to facilitate checkpointing.
-        let mut gas_params = GasParams {
-            gas_limit: BLOCK_GAS_LIMIT,
-            gas_fee_cap: self.gas_fee_cap.clone(),
-            gas_premium: self.gas_premium.clone(),
-        };
+            // TODO: Maybe we should implement something like the Ethereum facade for estimating fees?
```

Comment:
IIRC, we discussed this in some previous PR, I think always including validator messages is a good idea. I don't think they should be competing against other user transactions for block space.

Reply:
Yes I agree, that seems to cut to the heart of the problem much better. The fee estimation would be accidental complexity.